src/postgraphql/graphiql/src/components/PostGraphiQL.js

```diff
@@ -0,0 +1,195 @@
+   * update the navigation stack with this function.
+   *
+   * I’m sorry Lee Byron.
+   */
+  // TODO: Submit a PR which adds this as a non-hack.
+  _updateGraphiQLDocExplorerNavStack (nextSchema) {
+    // Get the documentation explorer component from GraphiQL. Unfortunately
+    // for them this looks like public API. Muwahahahaha.
+    const { docExplorerComponent } = this.graphiql
+    const { navStack } = docExplorerComponent.state
+
+    // If one type/field isn’t find this will be set to false and the
+    // `navStack` will just reset itself.
+    let allOk = true
+
+    // Ok, so if you look at GraphiQL source code, the `navStack` is made up of
+    // objects that are either types or fields. Let’s use that to search in
+    // our new schema for matching (updated) types and fields.
+    const nextNavStack = navStack.map((typeOrField, i) => {
+      // If we are not ok, abort!
+      if (!allOk)
+        return null
+
+      // If this is a type, let’s do some shenanigans...
+      if (isType(typeOrField)) {
+        // Let’s see if we can get a type with the same name.
+        const nextType = nextSchema.getType(typeOrField.name)
+
+        // If there is no type with this name (it was removed), we are not ok
+        // so set `allOk` to false and return undefined.
+        if (!nextType) {
+          allOk = false
+          return null
+        }
+
+        // If there is a type with the same name, let’s return it! This is the
+        // new type with all our new information.
+        return nextType
+      }
+      // If you thought this function was already pretty bad, it’s about to get
+      // worse. We want to update the information for an object field.
+      else {
+        // Ok, so since this is an object field, we will assume that the last
+        // element in our stack was an object type.
+        const nextLastType = nextSchema.getType(navStack[i - 1] ? navStack[i - 1].name : null)
+
+        // If there is no type for the last type in the nav stack’s name.
+        // Panic!
+        if (!nextLastType) {
+          allOk = false
+          return null
+        }
+
+        // If the last type is not an object type. Panic!
+        if (!(nextLastType instanceof GraphQLObjectType)) {
+          allOk = false
+          return null
+        }
+
+        // Next we will see if the new field exists in the last object type.
+        const nextField = nextLastType.getFields()[typeOrField.name]
+
+        // If not, Panic!
+        if (!nextField) {
+          allOk = false
+          return null
+        }
+
+        // Otherwise we hope very much that it is correct.
+        return nextField
+      }
+    })
+
+    // This is very hacky but works. React is cool.
+    this.graphiql.docExplorerComponent.setState({
+      // If we are not ok, just reset the `navStack` with an empty array.
+      // Otherwise use our new stack.
+      navStack: allOk ? nextNavStack : [],
+    })
```

Comment:
In the long term it would be better if `<GraphiQL />` took care to reset its own navStack (and related dependencies) when the `schema` prop changes; but I doubt we have time to raise a PR to do that right now!